src/main/java/org/terasology/structureTemplates/events/SpawnStructureEvent.java

```diff
@@ -15,15 +15,15 @@
  */
 package org.terasology.structureTemplates.events;
 
-import org.terasology.entitySystem.event.Event;
+import org.terasology.entitySystem.event.AbstractConsumableEvent;
 import org.terasology.structureTemplates.components.SpawnBlockRegionsComponent;
 import org.terasology.structureTemplates.util.transform.BlockRegionTransform;
 
 /**
  * When sent to entities with components like {@link SpawnBlockRegionsComponent} then the structure described by
  * that entity will be generated.
  */
-public class SpawnStructureEvent implements Event {
+public class SpawnStructureEvent extends AbstractConsumableEvent {
```

Comment:
My intention was to allow changing the "spawning process" (block placement) by just attaching a component. Although it is possible to define the grow structure component by hand (with more detailed/better build steps) the idea was to just reuse the existing template and place it in a specific order.

Maybe the pipeline in StructureTemplates should be adapted to allow for a more free definition of the spawning process? Keep SpawnStructureEvent consumable, and once it is spawned send a new StructureSpawnedEvent (or FinishedSpawnStructureEvent) to act on further steps?